### PR TITLE
Add driver type information in service-client telemetry

### PR DESCRIPTION
**Pull request by sonalideshpandemsft**
This change adds `driver`, `driverEndpointName` and `testVariant` information to service-client telemetry. Rest of the field such as `clientType`, `loaderId` and `containerId` are not displayed as part of `Test_start` and `Test_end` event name

[ADO Work Item](https://dev.azure.com/fluidframework/internal/_workitems/edit/8762)

### CODE REVIEW

File: packages/service-clients/end-to-end-tests/azure-client/src/test/.mocharc.cjs [hunk 1]
```diff
@@ -8,4 +8,24 @@
...
+	const endpointName = endpointIndex !== -1 ? args[endpointIndex + 1] : "";
+
+	return `${testDriver}-${endpointName}`;
+}
+
+function getFluidTestMocha(packageDir, additionalRequiredModules = []) {
+	const testVariant = getFluidTestVariant();
+	process.env.FLUID_TEST_VARIANT = testVariant;
+
+	return config(packageDir, additionalRequiredModules, testVariant);
```
Comment (Abe27342) [inline packages/service-clients/end-to-end-tests/azure-client/src/test/.mocharc.cjs:28]: this looks wrong, seems like you want `getFluidTestMochaConfig(packageDir, additionalRequiredModules, testVariant)` here. Same in other mocharc

File: packages/service-clients/end-to-end-tests/odsp-client/package.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
 		"test": "npm run test:realsvc:odsp:run",
 		"test:coverage": "c8 npm test",
 		"test:realsvc:odsp": "cross-env npm run test:realsvc:odsp:run",
-		"test:realsvc:odsp:run": "mocha --recursive \"lib/test/**/*.spec.*js\" --exit --timeout 20000",
+		"test:realsvc:odsp:run": "mocha --recursive \"lib/test/**/*.spec.*js\" --exit --timeout 20000 --config src/test/.mocharc.cjs -- --driver=odsp --driverEndpoint=odsp",
```
Comment (Abe27342) [inline packages/service-clients/end-to-end-tests/odsp-client/package.json:33]: Is there a reason for the `--` (i.e. not passing `--driver=odsp` directly to mocha)?

AFAIK that syntax is mostly just a thing for npm, where generally speaking, command-line arguments to `npm` are not plumbed through to child processes it creates. So e.g. `npm run foo --someFlag` runs the script `foo` and passes `someFlag` to the `npm` process, but whatever code executes as a result of running the `foo` script won't see `--someFlag` in its arguments. However `npm run foo -- --someFlag` is the opposite: the npm process doesn't have `--someFlag` but the script for `foo` will.

File: packages/service-clients/end-to-end-tests/azure-client/src/test/.mocharc.cjs (hunk 1 above)
Comment (sonalideshpandemsft) [reply packages/service-clients/end-to-end-tests/azure-client/src/test/.mocharc.cjs:28]: thanks for catching that!

**Comment by sonalideshpandemsft** [comment]
closing in favor of https://github.com/microsoft/FluidFramework/pull/22964

**Pull request closed without merging**